.cirrus.yml

```diff
@@ -37,6 +47,20 @@ redhat_egg_task:
         matrix:
           - image: fedora:35
           - image: fedora:34
+          - image: registry.access.redhat.com/ubi9-beta/ubi
+
+redhat8_egg_task:
+    egg_script:
+       - dnf -y install python3 python3-pip
+       - pip3 install "setuptools==40.6.3"
+       - python3 setup.py bdist_egg
+       - mv dist/avocado_framework-*egg /tmp
+       - python3 setup.py clean --all
+       - python3 -c 'import sys; import glob; sys.path.insert(0, glob.glob("/tmp/avocado_framework-*.egg")[0]); from avocado.core.main import main; sys.exit(main())' run /bin/true
+       - cd /tmp
+       - python3 -c 'import sys; from pkg_resources import require; require("avocado-framework"); from avocado.core.main import main; sys.exit(main())' run /bin/true
```

Comment:
I know those six lines are from `redhat_egg_task` and `debian_egg_task`. But since now we have three copies of the same block (lines 56-61), I wouldn't mind if this would be part of a Makefile target (or any other location), avoiding code duplication.

It would be much appreciated, helping us with the maintainability, readability, and reproducibility aspect.

Reply:
I wouldn't bother because if we add that in a makefile, we'll have to install make everywhere, this means still having separate jobs with their respectives dnf/apt-get. 
If you want to simplify things, the easiest is to stop using cirrus and move these jobs to github actions where it's easier to share the same job and add conditionals on the steps.

Reply:
> I wouldn't bother because if we add that in a makefile, we'll have to install make everywhere, this means still having separate jobs with their respectives dnf/apt-get.

We already have it as part of 'avocado-ci-fedora-' images and we are using it in one of the tasks. Since it is available on the other repositories and has a small footprint I would say it is not a big deal having it. It is important, to reduce the amount of duplicated lines whenever possible. 

> If you want to simplify things, the easiest is to stop using cirrus and move these jobs to github actions where it's easier to share the same job and add conditionals on the steps.

I believe this is more than just "simplifying things" and the CI it self. Besides the reasons already mentioned, it is about giving good examples for new contributors, avoiding big refactoring tasks in the future.

While there is support for poor conditionals and DSL, the YAML's declarative nature makes it difficult to read and write code to it. These DSLs are "hacks" IMO, and we should avoid them to ease maintenance and code inspection.

This goes for any CI that uses YAML as syntax and because of that, I believe we should be cautious about using code on it. For this reason, Cirrus supports Starlark. And although this could possibly be avoided there, I don't think it's necessary since we have our scripts and targets.

For now, I believe that having this inside Makefiles or even scripts would be enough. But in case you think this will add extra work to this PR, don't worry, I can work on it once merged.

Reply:
> I wouldn't bother because if we add that in a makefile, we'll have to install make everywhere, this means still having separate jobs with their respectives dnf/apt-get. If you want to simplify things, the easiest is to stop using cirrus and move these jobs to github actions where it's easier to share the same job and add conditionals on the steps.

Because those work as "quick deployment" checks, I second the idea of *not* requiring `make` installed.

Reply:
> > I wouldn't bother because if we add that in a makefile, we'll have to install make everywhere, this means still having separate jobs with their respectives dnf/apt-get.
> 
> We already have it as part of 'avocado-ci-fedora-' images and we are using it in one of the tasks. Since it is available on the other repositories and has a small footprint I would say it is not a big deal having it. It is important, to reduce the amount of duplicated lines whenever possible.
> 

The goals of those are very different.  These are "deployment" checks, that is, will Avocado run on fresh environments, with the minimum requirements?  The `avocado-ci-fedora*` are environments with a whole lot of extra dependencies installed, so they break the premise of these checks.

> > If you want to simplify things, the easiest is to stop using cirrus and move these jobs to github actions where it's easier to share the same job and add conditionals on the steps.
> 
> I believe this is more than just "simplifying things" and the CI it self. Besides the reasons already mentioned, it is about giving good examples for new contributors, avoiding big refactoring tasks in the future.
> 
> While there is support for poor conditionals and DSL, the YAML's declarative nature makes it difficult to read and write code to it. These DSLs are "hacks" IMO, and we should avoid them to ease maintenance and code inspection.
> 
> This goes for any CI that uses YAML as syntax and because of that, I believe we should be cautious about using code on it. For this reason, Cirrus supports Starlark. And although this could possibly be avoided there, I don't think it's necessary since we have our scripts and targets.
> 
> For now, I believe that having this inside Makefiles or even scripts would be enough. But in case you think this will add extra work to this PR, don't worry, I can work on it once merged.

The compromise solution is to put those in `.sh` files.  Something like `selftests/deployment/$platform.sh` or the like.